mu.rb

```diff
@@ -26,13 +26,25 @@ def log
     @log
   end
 
-  def query(query)
+  def query(query, **options)
     log.info "Executing query: #{query}"
-    sparql_client.query query
+    @sparql_client = sparql_client(**options)
+    @sparql_client.query query
   end
 
-  def sparql_client
-    options = {}
+  def sparql_client(**options)
+    if Mu::truthy? options[:sudo]
+      if Mu::truthy? ENV['ALLOW_MU_AUTH_SUDO']
+        options[:headers] = { 'mu-auth-sudo': 'true' }
+      else
+        log.error "Error, sudo request but service lacks ALLOW_MU_AUTH_SUDO header"
+      end
+    end
+    if options[:scope]
+      options[:headers] = { 'mu-auth-sudo': options[:scope] }
```

Comment:
Header here should be `mu-auth-scope`  (if I compare with https://github.com/mu-semtech/mu-javascript-template/commit/56e185e7081aff0061fa6c2f14061b2ffdadf024)

Reply:
Nice catch!